src/platform/android/DnssdImpl.cpp

```diff
@@ -316,7 +316,18 @@ void HandleResolve(jstring instanceName, jstring serviceType, jstring hostName,
     DnssdService service = {};
     CopyString(service.mName, jniInstanceName.c_str());
     CopyString(service.mHostName, jnihostName.c_str());
-    CopyString(service.mType, jniServiceType.c_str());
+
+    std::string serviceTypeStr(jniServiceType.c_str());
+
+    size_t index = serviceTypeStr.find(".");
```

Comment:
This code seems to be present in 2 separate places and looks identical. Lets place it in a stand alone function, like `CHIP_ERROR extractProtocol(const char *serviceType, DnssdServiceProtocol &out_protocol)`.

Reply:
Thanks for your review my code. I modify to add function on your comment.